content/6_Advanced/SPneg.mdx

```diff
@@ -362,6 +362,57 @@ public class CyclesFinding {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+INF = 10**18  # float('inf') will result in WA for some Test Cases
```

Comment:
wrong answer? you mean tle?

Reply:
No, I mean Wrong Answer

Reply:
The answer should be Yes along with the nodes but it results in straight No.

Reply:
hm
that seems a bit unintuitive, do you know why?

Reply:
not really, I haven't put much thought into it.

Reply:
got it .... it's because of weight constraint [-1e9, 1e9]

Reply:
I think the comment shouldn't be appropriate.

Reply:
the thing is that float('inf') doesn't give any real value number. so in the `if` condition, it will not pass through it. 
for that, we need some real value.

Reply:
using `sys.maxsize` works. do you want me to use that.

Reply:
Might be worth just defining a large constant.

Also the reason why `float('inf')` doesn't work is that any value plus `inf` is just `inf`.

`float('inf') + 10 = float('inf')`

Reply:
yes this is the issue with some TCs.

Reply:
> Also the reason why `float('inf')` doesn't work is that any value plus `inf` is just `inf`.

i mean yeah, that's kinda just how infinity works

Reply:
@SansPapyrus683 is it better to use `sys.maxsize`

Reply:
no i feel we need to figure out why float inf doesn't work

Reply:
he just mentioned that float inf + any number is still inf
like if the code breaks if that happens (which is expected behavior math-wise), i feel like something's wrong

Reply:
even if we add this condition `(w < 0 and dist[u] == float('inf') and dist[v] == float('inf'))` alongside the other one.
we still cannot overcome the next statement, which is to update the dist value which will still be infinity and that's where it is going wrong.

Reply:
idk don't most people just define a really big constant for INF anyways lol

Reply:
> which is to update the `dist` value which will still be infinity and that's where it is going wrong.

ok but it _should_ be infinity, no? unless the algorithm itself relies on a big number instead of actual infinity

Reply:
No, the updated value should be less than infinity then only it will work.  This is exactly where it is messed up.

Reply:
Unless it is some real number it is not going to work.

Reply:
> > which is to update the `dist` value which will still be infinity and that's where it is going wrong.
> 
> ok but it _should_ be infinity, no? unless the algorithm itself relies on a big number instead of actual infinity

My understanding is that it should still be updated because $w$ can be negative. So, the condition $\text{dist}_u + w < \text{dist}_v$ wouldn't work as intended.

Reply:
we are all coming to the same conclusion: `float('inf')` cannot be used here bcz it will cause trouble in updating `dist` values

Reply:
![image](https://github.com/user-attachments/assets/e1a2d956-fe4f-46f4-b5e9-18dce61246e9)

Reply:
I already did that with this condition but it resulted in WA for some other test cases.
![Screenshot from 2024-10-29 10-10-19](https://github.com/user-attachments/assets/e24c2965-9a28-4ef4-9c51-c77385a32dc3)
![Screenshot from 2024-10-29 10-10-39](https://github.com/user-attachments/assets/f6b462cf-3c19-4796-97b7-42a533255f30)